clustermesh/clustermesh.go

```diff
@@ -156,6 +157,39 @@ func (k *K8sClusterMesh) apiserverImage() string {
 	return utils.BuildImagePath(k.params.ApiserverImage, defaults.ClusterMeshApiserverImage, k.params.ApiserverVersion, k.imageVersion)
 }
 
+func (k *K8sClusterMesh) etcdImage() string {
+	etcdVersion := "v3.4.13"
+	arch := runtime.GOARCH
```

Comment:
This will give us the `GOARCH` for the host the Cilium CLI is running. I think we instead should to check the `GOARCH` for the cluster where Cilium is deployed.

Reply:
Because we use the `quay.io/coreos/etcd: 3.4.13` that it isn't multi-arch image. So k8s will pull `amd64` image. It can't run on  arm64 server. And i have got your point. Do you think adding param to the `K8sClusterMesh.params` is a good idea?

Reply:
I don't think this needs a dedicated parameter. Instead we should be able to get the target architecture from the k8s server version information (`apimachinery/pkg/version.Info` in the `Platform` field which contains the `GOOS`/`GOARCH` pair). This is already queried here:

https://github.com/cilium/cilium-cli/blob/f277e6540e7607d67d4616b90f19eed6e22fb018/internal/k8s/client.go#L555-L561

so I think you could adapt that method (and all its callers) to return the `version.Info` struct instead and then extract the arch from there.

Reply:
Okay, thank you. I will fix it later.